sqliterepo/cache.c

```diff
@@ -32,8 +33,29 @@ License along with this library.
 
 #define BEACON_RESET(B) do { (B)->first = (B)->last = -1; } while (0)
 
-volatile gboolean oio_sqlx_cache_fail_on_heavy_load = SQLX_HEAVYLOAD_FAIL;
-volatile gboolean oio_sqlx_cache_alert_on_heavy_load = SQLX_HEAVYLOAD_ALERT;
+static OIO_VAR_DEFINE_CONFIG(gboolean, _cache_fail_on_heavy_load,
+		"sqliterepo.cache.heavyload.fail", SQLX_HEAVYLOAD_FAIL);
+
+static OIO_VAR_DEFINE_CONFIG(gboolean, _cache_alert_on_heavy_load,
+		"sqliterepo.cache.heavyload.alert", SQLX_HEAVYLOAD_ALERT);
+
+static OIO_VAR_DEFINE_CONFIG(guint32, _cache_max_waiting,
+		"sqliterepo.cache.waiting.max", SQLX_MAX_WAITING);
+
+static OIO_VAR_DEFINE_CONFIG(guint, _cache_timeout_open,
+		"sqliterepo.cache.timeout.open", DEFAULT_CACHE_OPEN_TIMEOUT);
+
+static OIO_VAR_DEFINE_CONFIG(guint32, _heat_threshold,
+		"sqliterepo.heat.threshold", 1);
+
+static OIO_VAR_DEFINE_CONFIG(gint64, _grace_delay_cool,
+		"sqliterepo.cache.ttl.cool", SQLX_GRACE_DELAY_COOL * G_TIME_SPAN_SECOND);
+
+static OIO_VAR_DEFINE_CONFIG(gint64, _grace_delay_hot,
+		"sqliterepo.cache.ttl.hot", SQLX_GRACE_DELAY_HOT * G_TIME_SPAN_SECOND);
+
+static OIO_VAR_DEFINE_CONFIG(gint64, _cache_period_cond_wait,
+		"sqliterepo.wait.quantum", G_TIME_SPAN_SECOND);
```

Comment:
I don't like the name _sqliterepo.wait.quantum_. Actually it's a polling delay for _sqliterepo.cache.timeout.open_.